src/tools/rados/rados.cc

```diff
@@ -2445,8 +2445,18 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
 
     bufferlist bl;
     if (nargs.size() > min_args) {
-      string val(nargs[min_args]);
-      bl.append(val);
+      if (omap_key.compare("order") == 0 ||
+          omap_key.compare("features") == 0 ||
+          omap_key.compare("size") == 0 ||
+          omap_key.compare("flags") == 0 ||
+          omap_key.compare("snap_seq") == 0 ) {
+        unsigned long long val;
+        sscanf(nargs[min_args],"%llu", &val);
```

Comment:
@tchaikov , I looked up it, and found it seems no method to translate a string to unsigned int or unsigned long long. Only int, long, and long long. Please correct me.